bokeh/charts/data_source.py

```diff
@@ -154,7 +154,7 @@ def __init__(self, label, data, attr_specs):
 
         """
         self.label = label
-        self.data = data.reset_index()
```

Comment:
This conflicts with L171 (resetting the index then trying to read the index value).